src/quest/QuestModel.cpp

```diff
@@ -1,61 +1,95 @@
 #include <utility\YiString.h>
 #include <datamodel\YiAbstractDataModel.h>
-#include "quest/QuestModel.h"
+#include "QuestModel.h"
+#include "QuestObjectiveModel.h"
 
-QuestModel::QuestModel(CYIString name, CYIString description) : CYIAbstractDataModel(0, 1)
+QuestModel::QuestModel(CYIString name, CYIString description) : CYIAbstractDataModel(1)
 {
-	this->name = name;
-	this->description = description;
+    Initialize(name, description);
+}
+
+void QuestModel::Initialize(CYIString name, CYIString description)
+{
+    m_name = name;
+    m_description = description;
 }
-//
-//QuestModel::QuestModel(CYIString name, CYIString description, YI_INT32 nObjectives) : CYIAbstractDataModel(nObjectives, 1)
-//{
-//	this->name = name;
-//	this->description = description;
-//}
 
 QuestModel::~QuestModel()
-{	
+{
 }
 
 CYIString QuestModel::ToString() const
 {
-	//Quest Info
-	CYIString result;
-	result.Append(name);
-	result.Append(CYIString("\n"));
-	result.Append(description);
+    CYIString questInfo;
+    questInfo.Append(m_name);
+    questInfo.Append(CYIString("\n"));
+    questInfo.Append(m_description);
+
+//    YI_INT32 count = GetRowCount();
 
-	return result;
+    for (YI_INT32 i = 1; i < GetRowCount(); i++) //Skipping the first one because it is empty. Design flaw.
+    {
+        CYIAny data(GetItemData(GetIndex(i, 0)));
+
+        if (!data.Empty())
+        {
+            CYISharedPtr<QuestObjectiveModel> objective = data.Get<CYISharedPtr<QuestObjectiveModel>>();
+
+            questInfo.Append("\n");
+            questInfo.Append(objective->ToString());
+        }
+    }
+
+    return questInfo;
 }
 
 void QuestModel::AddObjective(CYIString name, const std::vector<CYIString> resolutions)
 {
-	InsertRow(GetRowCount() - 1);
-	
-	
-}
+    InsertRow(GetRowCount());
```

Comment:
you don't need to address this now, but maybe we can create our own DataModel class that would have a method call InsertLast? later....